ckanext/datastore/formats.py

```diff
@@ -0,0 +1,209 @@
+# encoding: utf-8
+from __future__ import annotations
+
+from ckanext.datastore.interfaces import IDatastoreDump
+
+from io import StringIO, BytesIO
+
+from contextlib import contextmanager
+from typing import Any, Optional, Union
+from simplejson import dumps
+
+from xml.etree.cElementTree import Element, SubElement, ElementTree
+
+import csv
+
+from codecs import BOM_UTF8
+
+
+BOM = "\N{bom}"
+
+
+class CSV(IDatastoreDump):
+    def get_format(self):
+        return 'csv'
+
+    def get_content_type(self):
+        return b'text/csv; charset=utf-8'
+
+    def write_records(self, records: list[Any]) -> bytes:
+        self.output.write(records)  # type: ignore
+        self.output.seek(0)
+        output = self.output.read().encode('utf-8')
+        self.output.truncate(0)
+        self.output.seek(0)
+        return output
+
+    @contextmanager
+    def get_writer(self, fields: list[dict[str, Any]], bom: bool = False):
```

Comment:
We need a way for a site to choose that the bom is set to True for these, so it could be a parameter to `__init__` instead